### PR TITLE
Hooking in project settings and team assignment

### PR DESCRIPTION
**Pull request by jcontonio**
* Re-structures project settings into separate containers to keep things clean
* Hooks in ProjectInfo updates to the server
* Hooks in ProjectTeam creation, update, and removal from the server
* Adds ability to disable dropdowns

### CODE REVIEW

**Comment by jcontonio** [comment]
I need help with this, as the API is 403'ing, requiring `roleId` in the body on create, which I am passing
I can remove people from projects
I can update roles

File: web/src/client/js/components/ProjectSettings/ProjectSettingsInfo/ProjectSettingsInfoContainer.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+const ProjectSettingsInfoContainer = ({ location }) => {
+  const { data: project } = useDataService(currentResource('project', location.pathname), [
+    location.pathname
+  ])
+
+  if (!project) return null
+
+  const debouncedUpdateHandler = debounce(1000, (body) => {
+    console.info('update project', body)
```
Comment (jjidt) [inline web/src/client/js/components/ProjectSettings/ProjectSettingsInfo/ProjectSettingsInfoContainer.js]: do you want this in here still?

File: web/src/shared/constants.js [hunk 2]
```diff
@@ -4,10 +4,16 @@ const FIELD_TYPES = {
...
+  ADMIN: 1,
+  CONTRIBUTOR: 2,
+  READER: 3
+}
+
 const ROLE_NAMES = {
-  READ: 'Reader',
-  WRITE: 'Contributor',
-  ADMIN: 'Admin'
+  [ROLE_IDS.ADMIN]: 'Admin',
```
Comment (jjidt) [inline web/src/shared/constants.js:14]: nice

**Review by jjidt — COMMENTED** [review]
Pushed up some changes and updated the docs, everything seems to be working as expected, just a couple comments

File: web/src/client/js/components/ProjectSettings/ProjectSettingsInfo/ProjectSettingsInfoContainer.js (hunk 1 above)
Comment (jcontonio) [reply web/src/client/js/components/ProjectSettings/ProjectSettingsInfo/ProjectSettingsInfoContainer.js]: Nope, removing.

**Review by jjidt — APPROVED** [review]

**Pull request merged**